New landing page

GuilhermehChaves opened the pull request:
Closes #2

GuilhermehChaves committed:
changing 'static' directory

creating 'style' directory into 'static'

GuilhermehChaves committed:
adding inital style

initial style, starting with responsive css

GuilhermehChaves committed:
starting the new landing page layout

GuilhermehChaves committed:
removing font files

adding 'Rubik' font via embed link

GuilhermehChaves committed:
starting intro section

GuilhermehChaves committed:
adding font 'JetBrains Mono' into 'static/font/'

GuilhermehChaves committed:
adding style for screens up to 490px

GuilhermehChaves committed:
finishing introduction section for screens up to 490px

GuilhermehChaves committed:
finishing mobile style

adding style to all mobile devices

GuilhermehChaves committed:
finishing desktop layout

adding responsive style for desktop screens

DennyLoko commented on index.html:
```diff
@@ -1,53 +1,348 @@
+
+  <script src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/316871/termynal.min.js"></script>
+  <script>
+    var termynal = new Termynal('#termynal');
```
It's possible to use a Termynal's config to start on-demand, so we can start the animation whenever the user scrolls to the element: https://github.com/ines/termynal#customising-termynal

DennyLoko commented on index.html:
```diff
@@ -1,53 +1,348 @@
+      }
+    }
+
+    function onResize() {
```
Hmmm... I don't think this is the correct form to do this. We can obtain the same result using CSS's [Media Queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries). W3 Schools has a nice article explaining this: https://www.w3schools.com/css/css_rwd_mediaqueries.asp

GuilhermehChaves replied:
It's because I researched how to put or remove an element in a div with CSS and I didn't find how only with Javascript

DennyLoko replied:
Ok, but you're not going to remove/add an element based on the screen size, you're going to hide/show this element. And you can achieve this using pure CSS...

DennyLoko replied:
Check the `Always Design for Mobile First` paragraph on the W3 Schools article I sent above, it shows how you would show elements when the user is accessing on screens larger than 768px.

GuilhermehChaves replied:
okay, it makes sense

DennyLoko commented on index.html:
```diff
@@ -1,53 +1,348 @@
+        </div>
+
+        <div class="code-image">
+          <img src="./static/img/code.svg"/>
```
We won't be using an image here, we'll be using the [Highlight.js](https://highlightjs.org/) for that, and will use the tabs on the left to change the content of the editor, similar to [this](https://root.co.za/card) site.

DennyLoko commented on index.html:
```diff
@@ -1,53 +1,348 @@
+
+  <hr>
+
+  <footer>
```
Looks like this footer doesn't matches to the layout... Can you take a look, please?

GuilhermehChaves committed:
refactoring css

creating new css classes and excluding repeated codes

GuilhermehChaves committed:
removing Javascript code

based on code review, removing Javascript code that adds/removes elements inside of another div, now show/hide elements only using CSS

GuilhermehChaves committed:
adding highlight.js

based on code review, adding highlight.js in place of the static image
to represent the API

GuilhermehChaves committed:
selecting options

selecting between bank transfer and bank slip

GuilhermehChaves committed:
correcting footer

based on code review, correcting footer according to the layout

GuilhermehChaves committed:
initing termynal

based on code review, initing termynal just when the user scrolls up to termynal element

GuilhermehChaves committed:
Replacing CSS with Sass

creating the mobile style with Sass

DennyLoko commented:
Esse PR voltou pra WIP, né @GuilhermehChaves ?

GuilhermehChaves committed:
Sass style for tablet

adjusting the tablet layout style

GuilhermehChaves committed:
desktop style

terminating Sass style for desktop devices

GuilhermehChaves committed:
widescreen style

ending stylization for widescreen devices

GuilhermehChaves committed:
repairing the title of the section 'account'

moving the title 150px for the right

DennyLoko commented on src/scss/base.scss:
```diff
@@ -0,0 +1,116 @@
+.btn,
+.border__btn,
+.intro__registered {
+    font-family: "JetBrains Mono"
```
Replace this font with Fira Mono and remove the JetBrains files and references, please.

GuilhermehChaves replied:
Okay 👍

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,326 @@
+            </div>
+
+            <div class="intro__access__registered">
+              <img src="./img/people.png" />
```
Can you remove the SVG file and also check for unused files and do a cleanup, please?

DennyLoko reviewed (changes requested):
Is this missing the mobile version? Because I'm not seeing the hamburger menu when on small devices.

GuilhermehChaves committed:
using Fira Mono font

based on code review removing JetBrains Mono font files and replacing with Fira mono

GuilhermehChaves committed:
removing unused files

based on code review, removing unused people.svg file

GuilhermehChaves commented:
> Is this missing the mobile version? Because I'm not seeing the hamburger menu when on small devices.

yeah, the hamburger menu is missing :/

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,326 @@
+      <div class="container">
+        <div class="card__content">
+          <h1 class="title">
+            Novo <span class="marked">CodeCard</span>. 💳 <span class="non-highlighted">Use débito ou crédito, você é
```
Looks like the content on this paragraph is wrong... Can you compare with the Figma project, please?

GuilhermehChaves replied:
I understood what happened it's because the text is different in mobile and desktop devices, which one I use?

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,326 @@
+      <nav>
+        <ul>
+          <li>
+            <button class="btn btn--header" type="button">Cadastrar</button>
```
Can you check the dimensions and font-size of this element, please?

DennyLoko replied:
Hmm, I see what happened here... You're using the `em` unit which scales the element size accordingly to the user screen, right?

GuilhermehChaves replied:
I'm using `rem` for font sizes, but in this case, the scale is a little bit like pixels `1.0rem = 10px` because the font size in HTML element is `62.5%`

DennyLoko commented on src/scss/base.scss:
```diff
@@ -0,0 +1,111 @@
+}
+
+.border {
+    border-radius: 2px;
```
The `border-radius` of the elements on the content isn't the same as on the button, the former should be `4px` instead of `2px`.

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,326 @@
+        </div>
+
+        <div class="border border--gray border--termynal">
+          <div class="codeline__border__termynal" data-termynal data-ty-typeDelay="70" data-ty-lineDelay="900">
```
Looks like the termynal is a little bit narrow than it should be...

DennyLoko replied:
Você chegou a olhar o tamanho do termynal? Ele está muito estreito, não está de acordo com o layout no figma...

GuilhermehChaves committed:
correcting the header button

based on code review, adjusting the header button dimensions and font size

GuilhermehChaves committed:
hamburger menu

creating a hamburger menu for mobile devices

GuilhermehChaves committed:
fixing up spacings

fixing up all incorrect spacings in elements

GuilhermehChaves committed:
changing termynal border-radius

changing border-radius from 2px to 4px

GuilhermehChaves committed:
fixing up the intro button

changing the button size from 64px to 56px

GuilhermehChaves committed:
removing padding from footer container

adding padding at the footer element

DennyLoko commented on src/scss/intro.scss:
```diff
@@ -0,0 +1,89 @@
+}
+
+@media (min-width: $tablet) {
+    .intro__access__registered {
```
Por que você está modificando o tamanho da imagem aqui? Isso está fazendo com que a imagem perca resolução...

DennyLoko commented on src/scss/api.scss:
```diff
@@ -0,0 +1,109 @@
+    }
+}
+
+.api__transactions__item__trace {
```
Faltou o border-radius aqui...

GuilhermehChaves committed:
adicionando border-radius

com base no code review adicionando border-radius ao .api__transactions__item__trace

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,357 @@
+      <nav class="nav__btn">
+        <ul>
+          <li>
+            <button class="btn btn--header" type="button">Cadastrar</button>
```
Esse botão tem que ter uma ação que leva o usuário para o formulário no rodapé, senti falta hoje enquanto estava testando no celular... O ideal seria fazer um scroll suave ao invés de dar aquele salto para o final.

DennyLoko replied:
Os botões estão em negrito lá no Figma, esquecemos de passar para cá...

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,357 @@
+              <div class="register__form-content__form-group">
+                <label for="cellphone" class="subtitle">Celular</label>
+                <div class="border border--input">
+                  <input class="border__input" id="cellphone" type="tel" placeholder="(__) _____ - ____"
```
Pode colocar uma máscara aqui? Para já ir preenchendo com a formatação à medida que a pessoa vai escrevendo...

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,357 @@
+              <div class="register__form-content__form-group">
+                <label for="email" class="subtitle">E-mail</label>
+                <div class="border border--input">
+                  <input class="border__input" id="email" type="email" placeholder="exemplo@codebank.com" name="email">
```
```suggestion
                  <input class="border__input" id="email" type="email" placeholder="exemplo@codebank.com.br" name="email">
```

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,357 @@
+            <li class="nav__item"><a href="#" class="nav__link">Sobre nós</a></li>
+          </div>
+
+          <div class="nav__menu__container">
```
Estou achando o X do menu muito grande e mais destacado que o logo do Codebank... Acho que rola de diminuir um pouco e usar um tom entre branco e cinza, para destacar menos...

![IMG_4A860201D6E8-1](https://user-images.githubusercontent.com/635014/92966427-e92e3480-f44d-11ea-950b-7915ecb12885.jpeg)

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,357 @@
+          <div class="nav__menu__container">
+            <li class="nav__item"> <span class="nav__title">Nossas redes</span></li>
+            <li class="nav__item"><a href="https://twitter.com/CodebankHQ" class="nav__link">Twitter</a></li>
+            <li class="nav__item"><a href="https://www.instagram.com/codebankhq/" class="nav__link">Instagram</a></li>
```
O link do Instagram não está aparecendo no Safari.

![IMG_4A860201D6E8-1](https://user-images.githubusercontent.com/635014/92966427-e92e3480-f44d-11ea-950b-7915ecb12885.jpeg)

GuilhermehChaves committed:
ajustando termynal

baseado no code review ajustando as dimensões do termynal

GuilhermehChaves committed:
tamanho original da imagem

removendo redimensionamento na imagem das pessoas cadastradas

GuilhermehChaves committed:
scroll para o formulário

com base no code review adicionando scroll para o formulário ao clicar
no botão

GuilhermehChaves committed:
máscara de telefone

com base no code review adicionando máscara no campo de telefone ao
digitar

GuilhermehChaves committed:
corrigindo placeholder

baseado no code review corrigindo texto do placeholder

GuilhermehChaves committed:
icone do menu hambúrguer

baseado no code review ajustando o ícone do menu hambúrguer

GuilhermehChaves committed:
adicionando borda no highlight.js

criando a borda cinza ao redor do código do highlight.js

GuilhermehChaves committed:
corrigindo link do menu habúrguer

com base no code review corrigindo link do instagram do menu hambúrguer
no safari do IOS

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,369 @@
+          <div class="container--api__code">
+              <div class="api__code">
+                <pre>
+                  <div class="border border--gray border--api__code">
```
A borda aqui no highlight.js ainda está errada no desktop e no mobile, vc chegou a dar uma olhada aqui?

GuilhermehChaves committed:
corrigindo borda do highlight

com base no code review, corrigindo borda do highlight.js

DennyLoko commented on src/index.html:
```diff
@@ -0,0 +1,369 @@
+                <pre>
+                  <div class="border border--gray border--api__code">
+                    <code class="javascript">// Faz o require da lib do Codebank e inicia com uma chave de teste
+                      const cb = require('@codebank/ted')('test_ghvbityg9086tftuhkgvtr8r7ityghuf');
```
O código está dando scroll horizontal, acho melhor colocar assim...

```suggestion
                      const API_KEY = 'test_ghvbityg9086tftuhkgvtr8r7ityghuf';
                      const cb = require('@codebank/ted')(API_KEY);
```

DennyLoko reviewed (changes requested):
Also, referente ao bug do ícone de fechar, descobri que o problema ocorre quando a gente roda `yarn run build`. Da uma olhada nesse bug, por favor?

GuilhermehChaves committed:
corrigindo ícone do menu hambúrguer

com base no code review, arrumando animação quebrada do ícone do menu

GuilhermehChaves committed:
alterando código no highlight.js

com base no code review, alterando código para evitar o scroll
horizontal
